sys/kern/vm_map.c

```diff
@@ -164,7 +164,8 @@ void vm_segment_destroy(vm_map_t *map, vm_segment_t *seg) {
 void vm_segment_destroy_range(vm_map_t *map, vm_segment_t *seg, vaddr_t start,
                               vaddr_t end) {
   assert(mtx_owned(&map->mtx));
-  assert(start >= vm_map_start(map) && end <= vm_map_end(map));
+  assert(seg->start >= vm_map_start(map) && seg->end <= vm_map_end(map));
```

Comment:
Is this assertion needed? Can't we assume that all segments in the `vm_map` are valid (i.e. they fall within the boundaries of the `vm_map`)?

Reply:
It is pretty obvious, idk why I had some doubts about it.